### PR TITLE
Ui/change colors

### LINKED ISSUE #80: Add ability to change track colors manually

### PR DESCRIPTION
**Pull request by BryonLewis**
Color Editing
Fixes #80 

Added in ability for users to edit types and have their own custom colors.  They can also edit type names.  Custom colors are saved in the meta data for the folder under `customTypeColors` and are a simple key/value pair for the type and the hex color code.

Chose to store the default colors as well if the user edits anything.  It seems like colors should be the same for all other items if the user refreshes the page.

### CODE REVIEW

File: client/src/components/TypeList.vue [hunk 1]
```diff
@@ -28,17 +65,145 @@ export default Vue.extend({
...
+              OK
+            </v-btn>
+          </v-card-actions>
+        </v-card>
+      </div>
+    </v-dialog>
   </div>
 </template>
+
+<style lang='scss'>
```
Comment (subdavis) [inline client/src/components/TypeList.vue:190]: used scoped style block.

File: client/src/components/TypeList.vue [hunk 2]
```diff
@@ -28,17 +65,145 @@ export default Vue.extend({
...
+.hover-show-parent {
+  .hover-show-child {
+    display: none;
+  }
+  &:hover {
+    .hover-show-child {
+      display: inherit;
+    }
+  }
+}
```
Comment (subdavis) [inline client/src/components/TypeList.vue:207]: Lint this `SCSS`.  We need to add automated linting, but for now, just fixing the tabbing will do.

File: client/src/use/useGirderDataset.ts [hunk 3]
```diff
@@ -72,6 +72,7 @@ export default function useGirderDataset() {
       throw new Error(`could not fetch dataset for id ${datasetId}`);
     }
     dataset.value = folder;
+    return dataset.value.meta;
```
Comment (subdavis) [inline client/src/use/useGirderDataset.ts:75]: It's kinda weird to load something then return a small part of it.  Maybe just return the whole thing.

File: client/src/views/TrackViewer/Viewer.vue [hunk 4]
```diff
@@ -100,11 +106,12 @@ export default defineComponent({
       allTypes,
       filteredTrackIds,
       enabledTrackIds,
+      updateTypeName,
     } = useTrackFilters({ trackMap, sortedTrackIds });
 
     // Initialize the view
     Promise.all([
-      loadDataset(datasetId),
+      loadDataset(datasetId).then((meta) => { loadTypeColors(meta && meta.customTypeColors); }),
```
Comment (subdavis) [inline client/src/views/TrackViewer/Viewer.vue:114]: I don't think `loadTypeColors` should be part of this try/catch.

If the REST request fails, we want to show the user an error dialog (TODO) but if `loadTypeColors` errors, that should dump to console.  Could this be moved after the catch using `dataset.value.meta` ?

File: client/src/views/TrackViewer/Viewer.vue [hunk 5]
```diff
@@ -244,6 +252,8 @@ export default defineComponent({
         featurePointing,
         featurePointingTarget,
       },
+      updateTypeColor,
+      updateTypeName,
```
Comment (subdavis) [inline client/src/views/TrackViewer/Viewer.vue:256]: Move these up into the `Methods used locally` section.

File: client/src/components/TypeList.vue [hunk 6]
```diff
@@ -19,6 +19,43 @@ export default Vue.extend({
...
+      }
+      this.colorRefresh();
+    },
+    /**
+     * Causes the color to refresh for annotations and events by toggling an item
+     */
+    colorRefresh() {
+      const val = this.checkedTypes.value.pop();
+      if (val) {
+        this.checkedTypes.value.push(val); // Causes a color refresh
```
Comment (subdavis) [inline client/src/components/TypeList.vue:55]: I don't think this is the right way to go about triggering an update.  

`typeColorMapper` should be a computed property dependent on the state of `dataset.meta`.  This only works for UI that rendered based on the state of `checkedTypes`, or in other words, the filtered list.  If we had a list of all tracks, or something else not derived from that, this wouldn't work and would probably confuse whoever was debugging it.

If you're busy I can take a swing at this.

Comment (BryonLewis) [reply client/src/components/TypeList.vue:55]: Yeah I was going to start yesterday moving it into a computed property but got sidetracked with some other stuff.  Most of today I'll probably working on other stuff.  I didn't like this way of doing it either and was trying to find a better way when I asked the other day.  Reminder that doing it this way means that the Layers.vue will probably need to watch the `typeColorMap` property to update the annotations in real time.

**Pull request merged**